core/sigagg/sigagg.go

```diff
@@ -43,47 +41,32 @@ func New(threshold int) *Aggregator {
 // into an aggregated signed duty data object ready to be broadcasted.
 type Aggregator struct {
 	threshold int
-	subs      []func(context.Context, core.Duty, core.PubKey, core.AggSignedData) error
+	subs      []func(context.Context, core.Duty, core.PubKey, core.SignedData) error
 }
 
 // Subscribe registers a callback for aggregated signed duty data.
-func (a *Aggregator) Subscribe(fn func(context.Context, core.Duty, core.PubKey, core.AggSignedData) error) {
+func (a *Aggregator) Subscribe(fn func(context.Context, core.Duty, core.PubKey, core.SignedData) error) {
 	a.subs = append(a.subs, fn)
 }
 
 // Aggregate aggregates the partially signed duty data for the DV.
-func (a *Aggregator) Aggregate(ctx context.Context, duty core.Duty, pubkey core.PubKey, parSigs []core.ParSignedData) error {
-	if len(parSigs) < a.threshold {
+func (a *Aggregator) Aggregate(ctx context.Context, duty core.Duty, pubkey core.PubKey, shares []core.ParSignedData) error {
+	if len(shares) < a.threshold {
 		return errors.New("require threshold signatures")
 	} else if a.threshold == 0 {
 		return errors.New("invalid threshold config")
 	}
 
-	// Get all partial signatures and one partial signed data object.
-	var (
-		blsSigs     []*bls_sig.PartialSignature
-		firstParSig core.ParSignedData
-		firstRoot   eth2p0.Root
-	)
-	for i, parSig := range parSigs {
-		root, err := getSignedRoot(duty.Type, parSig)
-		if err != nil {
-			return err
-		}
-		if i == 0 {
-			firstParSig = parSig
-			firstRoot = root
```

Comment:
removed root comparison, since it doesn't add much value. We will verify signatures in ParSigDB, and aggregation will fail if roots do not match.